From 1c67d0330ad9f52a9d8770430d32d93e9d278050 Mon Sep 17 00:00:00 2001
From: Jung Zhao <jung.zhao@rock-chips.com>
Date: Wed, 24 Jan 2018 15:03:52 +0800
Subject: [PATCH] video: rockchip: vpu: remove extra lock

when power off, no need to lock/unlock reset_lock. and
if set reg failed, need to unlock reset_lock.

Change-Id: If0a50920504bb69f9fa13790e85af1830fda7946
Signed-off-by: Jung Zhao <jung.zhao@rock-chips.com>
---
 drivers/video/rockchip/vcodec/vcodec_service.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/vcodec/vcodec_service.c b/drivers/video/rockchip/vcodec/vcodec_service.c
index 652a6440faa5..0746b84edd7c 100644
--- a/drivers/video/rockchip/vcodec/vcodec_service.c
+++ b/drivers/video/rockchip/vcodec/vcodec_service.c
@@ -822,14 +822,12 @@ static void vpu_service_power_off(struct vpu_service_info *pservice)
 
 	udelay(5);
 
-	mutex_lock(&pservice->reset_lock);
 	list_for_each_entry_safe(data, n, &pservice->subdev_list, lnk_service) {
 		if (data->mmu_dev && test_bit(MMU_ACTIVATED, &data->state)) {
 			clear_bit(MMU_ACTIVATED, &data->state);
 			vcodec_iommu_detach(data->iommu_info);
 		}
 	}
-	mutex_unlock(&pservice->reset_lock);
 	pservice->curr_mode = VCODEC_RUNNING_MODE_NONE;
 	if (pservice->hw_ops->power_off)
 		pservice->hw_ops->power_off(pservice);
@@ -1564,6 +1562,8 @@ static void reg_copy_to_hw(struct vpu_subdev_data *data, struct vpu_reg *reg)
 	} break;
 	default: {
 		vpu_err("error: unsupport session type %d", reg->type);
+
+		vcodec_exit_mode(data);
 		atomic_sub(1, &pservice->total_running);
 		atomic_sub(1, &reg->session->task_running);
 	} break;
-- 
2.35.3

